jsapp/js/components/submissions/submissionUtils.ts

```diff
@@ -531,11 +532,35 @@ export function getMediaAttachment(
   );
 
   submission._attachments.forEach((attachment) => {
+
     if (attachment.filename.includes(validFileName)) {
-      mediaAttachment = attachment;
+      // Check if the audio filetype is of type not supported by player and send it to format to mp3
+      if(attachment.mimetype.includes('audio/')
+        && !attachment.mimetype.includes('/mp3')
+        && !attachment.mimetype.includes('mpeg')
+        && !attachment.mimetype.includes('/wav')
+        && !attachment.mimetype.includes('ogg')
+      )
+      {
+        const lastItem = attachment.filename.split("/").pop();
+        const questionPath = Object.keys(submission).find(key => submission[key] === lastItem);
+
+        const newAudioURL = 
+        `${ROOT_URL}/api/v2/assets/${submission._xform_id_string}/data/${attachment.instance}/attachments/?xpath=${questionPath}&format=mp3`;
```

Comment:
When `questionPath` is `undefined`, this will crash the app.

Steps to break it:

1. Use a simple form with audio question 
[simple-audio-form.xlsx](https://github.com/kobotoolbox/kpi/files/12874349/simple-audio-form.xlsx)
2. Add submission using `flac` file (e.g. https://github.com/ietf-wg-cellar/flac-test-files/blob/main/subset/01%20-%20blocksize%204096.flac)
3. Go to Data Table and click this tiny play button. You will see `400` error on this:
   `GET
http://kf.kobo.local/api/v2/assets/aJpiarEu6gxjLrg7wuXpDq/data/34/attachments/?xpath=undefined&format=mp3` and another error: `Uncaught (in promise) DOMException: The media resource indicated by the src attribute or assigned media provider object was not suitable.`

Reply:
I'm not sure, but perhaps `Object.keys` above should be `Object.values`?

Reply:
So this problem is because when you upload the file it gets renamed by removing the '_' in the file. Unfortunately it's not as simple as just replacing the `_` with spaces because it only does it for the original file name but numeric values are getting added to the end of the file and that is our new file name.

For example: 
We upload: `01 - blocksize 4096.flac`
The attachment file name is `01_-_blocksize_4096-12_47_9.flac`
The question value is: `01 - blocksize 4096-12_47_9.flac`

We were comparing the file name to the question value in order to get the question path name but clearly that doesn't work when the name of the file is changing between the file name and the actual value of the question file. 
I'm working on the solution now and making sure theres not another, much cleaner way of doing this.